### PR TITLE
MindReading Part III

### PR DESCRIPTION
**Pull request by cyl19970726**

### CODE REVIEW

File: core/genesis.go [hunk 1]
```diff
@@ -439,14 +439,22 @@ func DefaultWeb3QTestnetGenesisBlock() *Genesis {
...
+			common.HexToAddress("0x560CD6c5054847c2AFCAae0dd9E34E22E6f47100"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
+			common.HexToAddress("0x3d2Bf29BaB6ec95422Eb143e856df3B0E775e50c"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
+			common.HexToAddress("0xb71af880FC87F2aA099251A2224ef3f38BE7f451"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
 		},
 		NextValidators: []common.Address{
-			common.HexToAddress("0x2cff0b8e36522eba76f6f5c328d58581243882e4"),
-			common.HexToAddress("0x959994471dee37411f579dd2820a8743cba20f46"),
-			common.HexToAddress("0x977cfc676bb06daed7ddfa7711bcfe8d50c93081"),
-			common.HexToAddress("0xcd21538af6e33ff6fcf1e2ca20f771413004cfd3"),
+			//common.HexToAddress("0x2cff0b8e36522eba76f6f5c328d58581243882e4"),
```
Comment (qzhodl) [inline core/genesis.go:449]: Is this change for test or you want to check it in?

File: params/config.go [hunk 2]
```diff
@@ -257,8 +257,10 @@ var (
 	}
 
 	Web3QTestnetValBootnodes = []string{
-		"/ip4/68.183.157.114/udp/33333/quic/p2p/12D3KooWEZ94qZgJgUNYiLwXahknkniYgozxw5eocijZJkew6Mj5",
-		"/ip4/128.199.102.174/udp/33333/quic/p2p/12D3KooWNjKALie7Cdpb4KG8axgABA4VCCsKoBQdvYvRhVZhbtPk",
+		//"/ip4/68.183.157.114/udp/33333/quic/p2p/12D3KooWEZ94qZgJgUNYiLwXahknkniYgozxw5eocijZJkew6Mj5",
+		//"/ip4/128.199.102.174/udp/33333/quic/p2p/12D3KooWNjKALie7Cdpb4KG8axgABA4VCCsKoBQdvYvRhVZhbtPk",
+		"/ip4/127.0.0.1/udp/33333/quic/p2p/12D3KooWEZ94qZgJgUNYiLwXahknkniYgozxw5eocijZJkew6Mj5",
```
Comment (qzhodl) [inline params/config.go:262]: Also is this change for local test?

File: params/config.go [hunk 3]
```diff
@@ -280,7 +282,7 @@ var (
 		PisaBlock:           big.NewInt(0),
 		ArrowGlacierBlock:   nil,
 		Tendermint: &TendermintConfig{
-			Epoch:                  1000,
+			Epoch:                  10080,
```
Comment (qzhodl) [inline params/config.go:285]: what is the reason for this change?

**Comment by cyl19970726** [comment]
The main responsibilities of this PR:
1. Add the mechanism of launching MindReading
2. Add the rpc interface of getMindReadingOutput

File: core/genesis.go [hunk 4]
```diff
@@ -439,14 +439,22 @@ func DefaultWeb3QTestnetGenesisBlock() *Genesis {
 		Alloc: map[common.Address]GenesisAccount{
 			common.HexToAddress("0x0E961a6A6235eFDB9a0F0BC753E395211B77cc28"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
 			common.HexToAddress("0x5C935469C5592Aeeac3372e922d9bCEabDF8830d"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
+			common.HexToAddress("0x024d6050275eec53b233B467AdA12d2C65B3AEce"): {Balance: new(big.Int).Mul(big.NewInt(1000000000000000000), big.NewInt(1000000000))}, // 1e9 Ether
```
Comment (qizhou) [inline core/genesis.go:442]: Need to revert these changes.

**Review by qizhou — APPROVED** [review]
A few minor comments.  The rest looks good.

**Review by qzhodl — APPROVED** [review]
